iothub/device/devdoc/API Design.md

```diff
@@ -6,297 +6,160 @@
 /// <summary>
 /// Retrieve the device properties.
 /// </summary>
+/// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
 /// <returns>The device properties.</returns>
-public Task<Properties> GetPropertiesAsync(CancellationToken cancellationToken = default);
+public Task<Properties> GetPropertiesAsync(CancellationToken cancellationToken);
 
 /// <summary>
 /// Update a single property.
 /// </summary>
 /// <param name="propertyName">Property name.</param>
 /// <param name="propertyValue">Property value.</param>
 /// <param name="componentName">The component name this property belongs to.</param>
-/// <param name="conventionHandler">A convention handler that defines the content encoding and serializer to use for the properties.</param>
+/// <param name="objectSerializer">A serializer to use for the properties.</param>
 /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
-public Task UpdatePropertyAsync(string propertyName, object propertyValue, string componentName = default, IConventionHandler conventionHandler = default, CancellationToken cancellationToken = default);
+public Task UpdatePropertyAsync(string propertyName, object propertyValue, string componentName = default, ObjectSerializer objectSerializer = default, CancellationToken cancellationToken = default);
```

Comment:
As track 2 about accepting an "ObjectSerializer" in public API.
Should we accept it during device client initialization (clientOptions)? -> this will tie it to the lifetime of the client and can't be changed per API call.

Reply:
This has been removed and placed inside of the PayloadCollection which removes it from the SDK surface and would need to be discovered to be implemented.